.github/workflows/pr-approvals.yaml

```diff
@@ -0,0 +1,42 @@
+# All PRs except trivial ones should require 2 approvals, since this a global setting
+# we cannot make this decision from github configuration alone. So we set the required
+# approved in github to 1 and make this check required which will enforce 2 approvals
+# unless overwritten or only CI files are touched.
+name: PR Approval Check
+"on":
+  pull_request:
+    branches: [main]
+jobs:
+
+  check_approvals:
+    name: Check for sufficient approvals
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Check for sufficient approvals
+        shell: bash --norc --noprofile {0}
+        env:
+          BODY: ${{ github.event.pull_request.body }}
+          GH_TOKEN: ${{ github.token }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+        run: |
+          echo "$BODY" | egrep -qsi '^disable-check:.*\<approval-count\>'
+          if [[ $? -ne 0 ]]; then
+            # Get the list of modified files in this pull request
+            files=$(gh pr view $PR_NUMBER --json files --jq '.files.[].path')
+            # Get the list of modified files in this pull request
+            approvals=$(gh pr view $PR_NUMBER --json reviews --jq '.reviews.[].state' | grep APPROVED | wc -l)
+
+            if [[ $approvals -lt 2 ]] && echo "${files}" | grep -Eq "^.github"; then
```

Comment:
It should be something like `echo ${files} | grep -vEq '^.github'`, right? If there are files changed outside of the github directory. I'd also do the grep inside jq here as well.

Reply:
moved the filter into the jq above